pkg/component/controller/etcd_member_reconciler.go

```diff
@@ -316,7 +365,25 @@ func (e *EtcdMemberReconciler) reconcileMember(ctx context.Context, member *etcd
 		return
 	}
 
-	if err = etcdClient.DeleteMember(ctx, memberID); err != nil {
+	err = retry.Do(func() error {
+		return etcdClient.DeleteMember(ctx, memberID)
+	},
+		retry.Delay(5*time.Second),
+		retry.LastErrorOnly(true),
+		retry.Attempts(5),
+		retry.Context(ctx),
+		retry.RetryIf(func(err error) bool {
+			// In case etcd reports unhealthy cluster, retry
+			msg := err.Error()
+			switch {
+			case strings.Contains(msg, "unhealthy cluster"):
```

Comment:
NB: There's probably more errors to retry, like "leader changed"

Reply:
That was the only error I saw during local testing